service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -330,4 +336,78 @@ void constructImputationInputsSuccess() {
     }
     assertEquals(totalInputs, allPipelineInputs.size());
   }
+
+  private static Stream<Arguments> mapInputTypeToCbasParameterTypeArguments() {
+    ParameterTypeDefinition stringParameterTypeResponse =
+        new ParameterTypeDefinitionPrimitive()
+            .primitiveType(PrimitiveParameterValueType.STRING)
+            .type(ParameterTypeDefinition.TypeEnum.PRIMITIVE);
+    ParameterTypeDefinition integerParameterTypeResponse =
+        new ParameterTypeDefinitionPrimitive()
+            .primitiveType(PrimitiveParameterValueType.INT)
+            .type(ParameterTypeDefinition.TypeEnum.PRIMITIVE);
+    ParameterTypeDefinition stringArrayParameterTypeResponse =
+        new ParameterTypeDefinitionArray()
+            .nonEmpty(true)
+            .arrayType(
+                new ParameterTypeDefinitionPrimitive()
+                    .primitiveType(PrimitiveParameterValueType.STRING)
+                    .type(ParameterTypeDefinition.TypeEnum.PRIMITIVE))
+            .type(ParameterTypeDefinition.TypeEnum.ARRAY);
+    return Stream.of(
+        // arguments: type specification, expected response
+        arguments(PipelineInputTypesEnum.STRING.toString(), stringParameterTypeResponse),
+        arguments(PipelineInputTypesEnum.VCF.toString(), stringParameterTypeResponse),
+        arguments(PipelineInputTypesEnum.INTEGER.toString(), integerParameterTypeResponse),
+        arguments(PipelineInputTypesEnum.STRING_ARRAY.toString(), stringArrayParameterTypeResponse),
+        arguments(PipelineInputTypesEnum.VCF_ARRAY.toString(), stringArrayParameterTypeResponse));
+  }
+
+  @ParameterizedTest
+  @MethodSource("mapInputTypeToCbasParameterTypeArguments")
+  void mapInputTypeToCbasParameterType(String inputType, ParameterTypeDefinition expectedResponse) {
+    assertEquals(expectedResponse, pipelinesService.mapInputTypeToCbasParameterType(inputType));
+  }
+
+  @Test
+  void prepareCbasWorkflowInputRecordLookupDefinitions() {
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>();
+    inputDefinitions.add(
+        new PipelineInputDefinition(
+            1L, "input1", PipelineInputTypesEnum.STRING.toString(), true, true, null));
```

Comment:
i would introduce some variability to the `true, true, null` parts of these definitions just to prove they arent actually important (i dont think they are at least)